ui.apps/src/main/content/jcr_root/apps/asset-share-commons/clientlibs/clientlib-site/js/storage.js

```diff
@@ -0,0 +1,262 @@
+            storage.removeItem(x);
+            localStorageEnabled = true;
+            return true;
+        }
+        catch(e) {
+            localStorageEnabled = false;
+            return e instanceof DOMException && (
+                // everything except Firefox
+                e.code === 22 ||
+                // Firefox
+                e.code === 1014 ||
+                // test name field too, because code might not be present
+                // everything except Firefox
+                e.name === 'QuotaExceededError' ||
+                // Firefox
+                e.name === 'NS_ERROR_DOM_QUOTA_REACHED') &&
+                // acknowledge QuotaExceededError only if there's something already stored
+                (storage && storage.length !== 0);
+        }
+    }
+
+    /* Return the local storage for Asset Share Commons as a JSON object */
+    function _getLocalStorage() {
+        if(_localStorageCheck()) {
+            return JSON.parse(window.localStorage.getItem(LOCAL_STORAGE_KEY)) || {};
+        }
+    }
+
+    /* Set the local Storage as a stringified JSON object */
+    function _updateLocalStorage(storageUpdate) {
+        if(_localStorageCheck()) {
+            window.localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(storageUpdate));
+        }
+    }
+
+    /* make a new request to get current user's profile */
+    function _initProfile() {
+
+        return new Promise(function(resolve, reject) {
+            var request = new XMLHttpRequest();
+            request.open('GET', CURRENT_USER_URI + '?nocache=' + new Date().getTime());
+            request.responseType = 'json';
+            request.onload = function() {
+                if (request.status === 200) {
+                    resolve(request.response);
+                }
+                else {
+                    reject(Error('profile did not load' + request.statusText));
+                }
+            };
+            request.onerror = function() {
+                // Also deal with the case when the entire request fails to begin with
+                // This is probably a network error, so reject the promise with an appropriate message
+                    reject(Error('There was a network error.'));
+             };
+            request.send();
+        });
+    }
+
+    /* dispatch event that profile was loaded sucessfully */
+    function _announceProfileLoaded(profile) {
+        var event = new CustomEvent(ns.Events.PROFILE_LOAD, {detail:profile}),
+            element = document.getElementsByTagName("body")[0];
+            profileLoaded = true;
+            element.dispatchEvent(event);
+    }
+
+    //initial call to retrieve profile
+    _initProfile().then(function(response) {
+        if(response.type === 'user') {
+           _setUserProfile(response);
+           profileLoaded = true;
+           _announceProfileLoaded(response);
+        }
+    });
+
+    return {
+        getReturnUrl: getReturnUrl,
```

Comment:
WDYT about creating sub-storage objects? 
- getCart()
- getProfile()
- getNavigation()

Feels a bit muddled having everything just on this big Storage object? This file is definitely alot to take in ..

Reply:
Yup agreed.  Are you thinking something like:

```js
AssetShare.Storage = (function (window, ns) {

       var cart = {...};
       var profile= {...};
       var navigation = {...};

       //some "global
       return {
           getCart : cart,
           getProfile: profile,
           getNavigation: navigation
         };

    } window, AssetShare)); 
```

Or break it into separate files?